tests/pipeline/pipeline_tests.py

```diff
@@ -102,21 +102,39 @@ def test_get_dag_definition(self):
                      'dag = DAG(dag_id=\'foo\', schedule_interval=\'bar\', '
                      'default_args=default_args)\n\n')
 
+  def test_get_datetime_expr(self):
+    import datetime
+    from pytz import timezone
+    datetime_expr = pipeline.Pipeline._get_datetime_expr('2009-05-05T22:28:15Z')
+    self.assertEqual(datetime_expr,
+                     'datetime.datetime.strptime(\'2009-05-05T22:28:15Z\', '
+                     '\'%Y-%m-%dT%H:%M:%SZ\').replace(tzinfo=timezone(\'UTC\'))')
+    self.assertEqual(eval(datetime_expr),
+                     datetime.datetime(2009, 5, 5, 22, 28, 15,
+                                       tzinfo=timezone('UTC')))
+
+    # Incorrectly formatted strings should throw (in this case without the 'Z')
+    with self.assertRaises(ValueError):
+      eval(pipeline.Pipeline._get_datetime_expr('2009-05-05T22:28:15'))
```

Comment:
This is actually testing that the code generated is valid python (i.e. it executes). It is generated based on user input. In this particular line, we are checking that it is not fact not valid in cases where the user input is not according to the specified format (i.e.  '%Y-%m-%dT%H:%M:%SZ').